examples/apps/cli/main.c

```diff
@@ -97,3 +98,22 @@ int main(int argc, char *argv[])
 
     return 0;
 }
+
+/*
+ * Provide, if required an "otPlatLog()" function
+ */
+#if ((OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_NONE) ||  (OPENTHREAD_CONFIG_LOG_OUTPUT == OPENTHREAD_CONFIG_LOG_OUTPUT_APP_STREAM))
+void otPlatLog(otLogLevel aLogLevel, otLogRegion aLogRegion, const char *aFormat, ...)
```

Comment:
You miss the fact that the NCP also produces logs and sends them up to the Linux host - so that they can be written into the Linux host syslog...

the intent here with this change is the CERT case is this:

a)   CERT_LOGS=1  - this enables the cert log
b)   Then - for the CLI example you need to turn on DEFAULT_LOG_OUTPUT=APP_STREAM

Currently in OpenThread there are two examples,  the CLI and the NCP

In the future there might be a different example - and it would provide its own "APP_STREAM"

maybe the name APP_STREAM is bad and it should be   APP_PROVIDED or something? 

The intent here is this:   *the application* provides the function - not the library.

In other cases, we want the generic feature in the platform to handle the log.